### DISCUSSION TITLE
Add queue lock to force running only one job at a time

### DISCUSSION THREAD

**Discussion by xhzhu0628** [McCloudS/subgen#52] [Ideas]
import threading
from queue import Queue

# start queue
task_queue = Queue()

# start queue lock
task_queue_lock = threading.Lock()

def transcription_worker():
    while True:
        task = task_queue.get()

        gen_subtitles(task['path'], task['transcribe_or_translate'], task['force'])

        task_queue.task_done()
        time.sleep(5*60)
        # show queue
        with task_queue_lock:
            print(f"There are {task_queue.qsize()} tasks left in the queue.")

# activate thread
threading.Thread(target=transcription_worker, daemon=True).start()

**Comment by McCloudS** [comment]
Queuing appears to mostly work for me, but I'm open to any PRs.  Whisper has a threads/workers option. Converting to discussion if you wanted to try a PR.  

``cpu_threads=whisper_threads, num_workers=concurrent_transcriptions``
